### DISCUSSION TITLE
Updates since the last time get events.

### DISCUSSION THREAD

**Discussion by dmitrysync** [ecederstrand/exchangelib#972] [General]
How do I get updates since the last time I tried to get events?

Microsoft graph uses `$deltatoken`.

`$deltatoken` -  A state token returned in the `deltaLink` URL of the previous delta function call for the same user collection, indicating the completion of that round of change tracking.

Does the `exchangelib` have the same functionality?

**Comment by ecederstrand** [comment]
You’re probably looking for the sync and notifications support. There are various options described at https://ecederstrand.github.io/exchangelib/#synchronization-subscriptions-and-notifications depending on your requirements.

**Reply by dmitrysync** [reply]
I saw this chapter.
But that isn't what I need.

**Reply by ecederstrand** [reply]
If subscription ID + watermark isn’t what you need, then you’ll need to elaborate on your requirements.

**Reply by dmitrysync** [reply]
Let's assume the following.

At the beginning of working with the calendar, I call `view()` and receive events

After a while, I call `view()` again and get only those events that have changed.

**Reply by ecederstrand** [reply]
That’s basically what `account.calendar.sync_items()` does. It doesn’t work exactly like `view()` but it gives you all events in the calendar folder since the last call to `sync_items()`.

**Reply by dmitrysync** [reply]
Maybe it will work.

But the important thing is that all calls must be made through the same connection.

**Reply by ecederstrand** [reply]
The same TCP connection? Nobody can really guarantee that, but exchangelib does reuse connections.
